drivers/sensor/ti/tmag5273/tmag5273.h

```diff
@@ -195,10 +195,13 @@
 
 /* Register DEVICE_ID */
 #define TMAG5273_VER_POS	0
-#define TMAG5273_VER_MSK	GENMASK(1, 0)
+#define TMAG3001_VER_POS        2
+#define TMAG5273_VER_MSK        GENMASK(3, 0)
```

Comment:
TMAG5273 datasheet specifies that only bits 0 and 1 are specified where 0b01 and 0b10 are used to distinguish between the X1 and X2 variants. The value of the other bits are not defined, but this change assumes that the other bits are 0.

Indeed, the TMAG5273A2 I'm using returns 0b00000110, so by masking more bits this doesn't match TMAG5273_VER_TMAG5273X2 anymore.

OTOH, TMAG3001 datasheet specifies that bits 0 and 1 are reserved, so they probably shouldn't be relied on to stay 0 either.

Reply:
You are right. Will you create a new pull request to fix it or should I?

Reply:
I might get to this in a few weeks. If you have the time sooner feel free to make the change.

If it were me, I'd create two compatibles and have this driver handle both, since 300x has a few other feature differences as well.

Reply:
Got around to doing this

https://github.com/zephyrproject-rtos/zephyr/pull/79955